From 288f54f0921a65f92be52f3624fb8ab2d33c2487 Mon Sep 17 00:00:00 2001
From: lion0738 <lion0738@naver.com>
Date: Sun, 10 Jan 2016 16:55:01 +0900
Subject: [PATCH 089/192] Snap: Set parameters before starting preview

Old parameters are used when grabbing focus mode.
Camera parameters should be set to FocusOverlayManager before starting preview.

Change-Id: I908559e1c2003be47b486996b0ec016b78107468
---
 src/com/android/camera/VideoModule.java | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index d1b39cb74..0119138fb 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -3085,12 +3085,6 @@ public class VideoModule implements CameraModule,
         mPreferences.setLocalId(mActivity, mCameraId);
         CameraSettings.upgradeLocalPreferences(mPreferences.getLocal());
         openCamera();
-        readVideoPreferences();
-        mUI.applySurfaceChange(VideoUI.SURFACE_STATUS.SURFACE_VIEW);
-        startPreview();
-        initializeVideoSnapshot();
-        resizeForPreviewAspectRatio();
-        initializeVideoControl();
 
         CameraInfo info = CameraHolder.instance().getCameraInfo()[mCameraId];
         boolean mirror = (info.facing == CameraInfo.CAMERA_FACING_FRONT);
@@ -3098,6 +3092,13 @@ public class VideoModule implements CameraModule,
         mFocusManager.setMirror(mirror);
         mFocusManager.setParameters(mParameters);
 
+        readVideoPreferences();
+        mUI.applySurfaceChange(VideoUI.SURFACE_STATUS.SURFACE_VIEW);
+        startPreview();
+        initializeVideoSnapshot();
+        resizeForPreviewAspectRatio();
+        initializeVideoControl();
+
         initializeCapabilities();
 
         // From onResume
-- 
2.17.1

